node/middlewares/redirect.ts

```diff
@@ -1,8 +1,19 @@
 import type { ServiceContext } from '@vtex/api'
 
 export async function redirect(ctx: ServiceContext, next: () => Promise<void>) {
-  // TODO: implement redirect function
-  console.log("🚀 ~ file: redirect.ts ~ line 4 ~ redirect ~ ctx", ctx)
+  const redirectState = ctx.state as State
+
+  const redirectUri = `https://${
+    redirectState.workspace && redirectState.workspace !== 'master'
```

Comment:
Can't you use optional chaining here ?
```suggestion
    redirectState?.workspace !== 'master'
```

Reply:
I believe you'd have to invert the condition for it to work, though 🤔 because with this suggestion, an `undefined` workspace would still be `!== master` and fall into the first condition, which is not desired.

With the current solution, undefined workspaces fall back into `''`, which is the expected result.

Reply:
Initially I thought the same thing, but unfortunately I tested it and it didn't work. Because the left side will return `undefined` and `undefined` is different from `master`, so it will be `true` instead of `false`. Therefore, it will return `undefined--{account}.myvtex.com` and not `{account}.myvtex.com`.
But it was a good point, thanks!

Reply:
The case where it didn't work was when the workspace parameter was not passed, just to be clear.